rasa/core/tracker_store.py

```diff
@@ -1236,3 +1234,132 @@ def _load_from_module_name_in_endpoint_config(
             f"Using `InMemoryTrackerStore` instead."
         )
         return InMemoryTrackerStore(domain)
+
+
+class AwaitableTrackerStore(TrackerStore):
+    """Wraps a tracker store so it can be implemented with async overrides."""
+
+    def __init__(
+            self,
+            tracker_store: TrackerStore,
+    ) -> None:
+        """Create a `AwaitableTrackerStore`.
+        Args:
+            tracker_store: the wrapped tracker store.
+        """
+
+        self._tracker_store = tracker_store
+
+        super().__init__(tracker_store.domain, tracker_store.event_broker)
+
+    @property
+    def domain(self) -> Optional[Domain]:
+        return self._tracker_store.domain
+
+    @domain.setter
+    def domain(self, domain: Optional[Domain]) -> None:
+        self._tracker_store.domain = domain
+
+    @staticmethod
+    def create(obj: Union["TrackerStore", EndpointConfig, None],
+               domain: Optional[Domain] = None,
+               event_broker: Optional[EventBroker] = None, ) -> Optional[TrackerStore]:
+        if isinstance(obj, TrackerStore):
+            return AwaitableTrackerStore(obj)
+        else:
+            return None
```

Comment:
Could be worth failing fast here and raising an error?

Reply:
Done!